pkg/ccl/backupccl/system_schema.go

```diff
@@ -242,12 +245,105 @@ func usersRestoreFunc(
 	return nil
 }
 
+func roleMembersRestoreFunc(
+	ctx context.Context,
+	execCfg *sql.ExecutorConfig,
+	txn *kv.Txn,
+	systemTableName, tempTableName string,
+) (retErr error) {
+	if !execCfg.Settings.Version.IsActive(ctx, clusterversion.V23_1RoleMembersTableHasIDColumns) {
+		return defaultSystemTableRestoreFunc(ctx, execCfg, txn, systemTableName, tempTableName)
+	}
+
+	// It's enough to just check if role_id exists since member_id was added at
+	// the same time.
+	hasIDColumnsQuery := fmt.Sprintf(`SELECT EXISTS (SELECT 1 FROM [SHOW COLUMNS FROM %s] WHERE column_name = 'role_id')`, tempTableName)
+	row, err := execCfg.InternalExecutor.QueryRow(ctx, "has-id-columns", txn, hasIDColumnsQuery)
+	if err != nil {
+		return err
+	}
```

Comment:
Depending on whose PR merges first, I made a small helper for this.

Reply:
Thanks, I've changed it to use your helper function!